File: frontend/src/app/modules/statistics/models/participant.ts
import {MentalState} from './mental-state';
import {Posture} from './posture';

export class Participant {
  id: number;
  nickname: string;
  posture: Posture;
  placeId: number;
  created: Date;
  leftGroup: Date;
  mentalStates: MentalState[];

  constructor(rawParticipant: Partial<Participant>) {
    Object.assign(this, rawParticipant);
    if (rawParticipant.mentalStates) {
      this.mentalStates = rawParticipant.mentalStates.map(mentalState => new MentalState(mentalState));
    }
  }
}

File: unity/Assets/Scripts/models/Participant.cs
namespace models
{
    public class Participant
    {
        public readonly string Nickname;
        public readonly Posture Posture;
        public readonly int PlaceId;

        public Participant(string nickname, Posture posture, int placeId = -1)
        {
            Nickname = nickname;
            Posture = posture;
            PlaceId = placeId;
        }
    }
}